libraries/AP_Mount/AP_Mount_Siyi.cpp

```diff
@@ -550,24 +560,31 @@ bool AP_Mount_Siyi::send_1byte_packet(SiyiCommandId cmd_id, uint8_t data_byte)
 void AP_Mount_Siyi::rotate_gimbal(int8_t pitch_scalar, int8_t yaw_scalar, bool yaw_is_ef)
 {
     // send lock/follow value if it has changed
-    if ((yaw_is_ef != _last_lock) || (_lock_send_counter >= AP_MOUNT_SIYI_LOCK_RESEND_COUNT)) {
-        set_lock(yaw_is_ef);
-        _lock_send_counter = 0;
-        _last_lock = yaw_is_ef;
-    } else {
-        _lock_send_counter++;
+    const GimbalMotionMode mode = yaw_is_ef ? GimbalMotionMode::LOCK : GimbalMotionMode::FOLLOW;
+    if (mode != _config_info.motion_mode) {
+        set_motion_mode(mode);
```

Comment:
I wonder if maybe set_motion_mode() should return a boolean and then we can immediately return if it fails.  It almost never fails but it might be good to do it this way anyway.

One slight concern is that the set_motion_mode() may be called many times in a row every time the lock/follow is changed because there will likely be a delay between when we call set_motion_mode() and when we next hear back from the gimbal on what its mode is via the ACQUIRE_GIMBAL_CONFIG_INFO message.  We should at least check how many times we end up calling this in a row and be sure that we think this is OK.

Reply:
Made `set_motion_mode()` return a `bool`. 

> One slight concern is that the set_motion_mode() may be called many times in a row

I ran some tests: 
* it was sending ~200 times in the initial implementation
* it was sending ~20 times if I sent a config request immediately after changing the mode
* it sends once if I manually set `_config_info.motion_mode` to the new `mode`